console-subscriber/src/aggregator.rs

```diff
@@ -431,6 +445,18 @@ impl Aggregator {
         }
     }
 
+    fn get_or_insert_task_id(&mut self, span_id: span::Id) -> TaskId {
+        match self.task_id_mappings.entry(span_id) {
+            Entry::Occupied(entry) => *entry.get(),
+            Entry::Vacant(entry) => {
+                let task_id = self.task_id_counter;
+                entry.insert(task_id);
+                self.task_id_counter = self.task_id_counter.wrapping_add(1);
+                task_id
+            }
+        }
```

Comment:
nit/TIOLI: I think this `match` could be replaced with [`Entry::or_insert_with`](https://doc.rust-lang.org/stable/std/collections/hash_map/enum.Entry.html#method.or_insert_with)
```suggestion
        self.task_id_mappings
            .entry(span_id)
            .or_insert_with(|| {
                let task_id = self.task_id_counter;
                entry.insert(task_id);
                self.task_id_counter = self.task_id_counter.wrapping_add(1);
                task_id
            })
```

Reply:
This was actually the first thing I tried, but unfortunately the borrow checker is not happy about the mutation inside the closure 😕 I'm going to a bit more reading to see if there are any other idiomatic ways.

```
error[E0500]: closure requires unique access to `self` but it is already borrowed
   --> console-subscriber/src/aggregator.rs:449:62
    |
449 |         *self.task_id_mappings.entry(span_id).or_insert_with(|| {
    |          ---------------------                -------------- ^^ closure construction occurs here
    |          |                                    |
    |          |                                    first borrow later used by call
    |          borrow occurs here
450 |             let task_id = self.task_id_counter;
451 |             self.task_id_counter = self.task_id_counter.wrapping_add(1);
    |             -------------------- second borrow occurs due to use of `self` in closure

```

Reply:
oh, huh, never mind — the current approach is fine, then, if the borrow checker doesn't like this.

I think we _could_ "split" the borrows, like this:
```suggestion
        let task_id_mappings = &mut self.task_id_mappings;
        let task_id_counter = &mut self.task_id_counter;
        task_id_mappings
            .entry(span_id)
            .or_insert_with(|| {
                let task_id = *task_id_counter;
                entry.insert(task_id);
                *task_id_counter = task_id_counter.wrapping_add(1);
                task_id
            })
```
but this is the same number of lines as the `match` version, and i'm not sure if it's meaningfully clearer. so, it may not be worth bothering...